### PR TITLE
Add CI to run tests

### PR DESCRIPTION
**Pull request by TimurSultanov**
Add CI to the repo to run tests

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#### Overview
+
+- Ticket: URL_HERE
+
+###### Optional
+
+Reference all core PRs for auto-link back-references.
+
+Core PRs
```
Comment (dereuromark) [inline .github/PULL_REQUEST_TEMPLATE.md]: Please remove those core internal things, like this file

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+jobs:
+  validation:
+    runs-on: ubuntu-18.04
+    strategy:
+      fail-fast: false
+      matrix:
+        php-version: [
+            '7.1',
+            '7.4'
```
Comment (dereuromark) [inline .github/workflows/ci.yml]: This should be min/max with latest 8.1 now

File: composer.json [hunk 3]
```diff
@@ -12,13 +12,11 @@
     ]
   },
   "require": {
-    "php": ">=7.1",
+    "php": "^7.1",
```
Comment (dereuromark) [inline composer.json]: Please use >= as per rules

File: composer.json [hunk 4]
```diff
@@ -12,13 +12,11 @@
...
     "propel/propel": "~2.0@dev"
   },
   "require-dev": {
-    "phpunit/phpunit": ">=7.0",
-    "codeclimate/php-test-reporter": "dev-master",
-    "athens/standard": "*"
+    "phpunit/phpunit": "^7.0"
```
Comment (dereuromark) [inline composer.json]: Should be also including up to 9 if possible for latest PHP8+

File: .github/workflows/ci.yml (hunk 2 above)
Comment (TimurSultanov) [reply .github/workflows/ci.yml]: it wasn't tested on php 8+, so I'm not sure it will work

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -0,0 +1,58 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+      - feature/[a-z]+-[0-9]+/dev-*
```
Comment (dereuromark) [inline .github/workflows/ci.yml]: branch filtering can be removed

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (TimurSultanov) [reply .github/PULL_REQUEST_TEMPLATE.md]: removed

File: .github/workflows/ci.yml (hunk 2 above)
Comment (TimurSultanov) [reply .github/workflows/ci.yml]: changed to 8.1

File: composer.json (hunk 3 above)
Comment (TimurSultanov) [reply composer.json]: done

File: composer.json (hunk 4 above)
Comment (TimurSultanov) [reply composer.json]: changed to "^7.0 || ^8.0 || ^9.0"

File: .github/workflows/ci.yml (hunk 5 above)
Comment (TimurSultanov) [reply .github/workflows/ci.yml]: removed

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,4 @@
+#### Overview
+
+- Ticket: URL_HERE
```
Comment (dereuromark) [inline .github/PULL_REQUEST_TEMPLATE.md]: The whole file can be removed IMO

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    strategy:
+      fail-fast: false
+      matrix:
+        php-version: [
+            '7.1',
+            '8.1'
+        ]
+
+    steps:
+      - uses: actions/checkout@v1
```
Comment (dereuromark) [inline .github/workflows/ci.yml]: We use newer template using v2

File: composer.json [hunk 8]
```diff
@@ -16,9 +16,7 @@
     "propel/propel": "~2.0@dev"
   },
   "require-dev": {
-    "phpunit/phpunit": ">=7.0",
```
Comment (dereuromark) [inline composer.json:19]: Can we use PSR4 class loading or is this not possible here?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (dereuromark) [reply .github/PULL_REQUEST_TEMPLATE.md]: "file"

File: .github/workflows/ci.yml (hunk 2 above)
Comment (dereuromark) [reply .github/workflows/ci.yml]: Thats what tests are for, and CI.

Comment (dereuromark) [reply .github/workflows/ci.yml]: Any reason why we are not going 7.3+?
Then also testing is easier, e.g. PHPUnit 9+ only.
And you can use our tools like code sniffer.
See https://spryker.atlassian.net/wiki/spaces/CORE/pages/3012034714/Standalone+libraries

Comment (TimurSultanov) [reply .github/workflows/ci.yml]: Well, it works with php 7.1 as well, does it make sense to limit the PHP version with 7.3+?

File: composer.json (hunk 8 above)
Comment (TimurSultanov) [reply composer.json:19]: I tried, tests failed with it

File: .github/workflows/ci.yml (hunk 2 above)
Comment (dereuromark) [reply .github/workflows/ci.yml]: We dont technically support any tool below this for the reasons stated above (not even code sniffer works here anymore which is always at least 1 minor behind).

**Review by dereuromark — APPROVED** [review]
We can release a 0.1 as is
And then fix up to 7.3+ afterwards.
We might also want to check PSR4 autoloading afterwards as all new libs should follow that standard.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (TimurSultanov) [reply .github/PULL_REQUEST_TEMPLATE.md]: the file removed

**Comment by matweew** [comment]
Please recheck README file, you can add original repository link and change the name, because `Athens` it's the name of the framework https://github.com/AthensFramework

File: .github/workflows/ci.yml (hunk 2 above)
Comment (dereuromark) [reply .github/workflows/ci.yml]: Note: Also propel itself will now be 7.3+ shortly.

**Comment by dereuromark** [comment]
Also all badges need to be fixed up to ours.

**Comment by dereuromark** [comment]
What the status here, can we merge and release this?

**Comment by TimurSultanov** [comment]
Hello, @dereuromark , we have a ticket to align this library with our standards: https://spryker.atlassian.net/browse/APPS-2536 
But we can merge this PR and create a new one for the ticket, what do you think?

**Pull request merged**